components/hybrid_engine/src/engine_iterator.rs

```diff
@@ -97,3 +99,53 @@ where
         }
     }
 }
+
+pub struct HybridEngineIterMetricsCollector<EK, EC>
+where
+    EK: KvEngine,
+    EC: RangeCacheEngine,
+{
+    collector: Either<
```

Comment:
Why do we need to have HybridEngineIterMetricsCollector again?
It seems to me as long as the caller of HybridEngineIterMetricsCollector has the MetricsExt instance, it can call the relevant APIs directly.

Reply:
Yes, understood. But for one HybridEngineIterMetricsCollector instance, it's either the RocksDB or RangeCache, which have the same interface. So why don't we use this same interface directly instead of a wrapper.